### ISSUE TITLE
rkt driver disabling of signature validation is too broad

### ISSUE THREAD

**Issue by blalor** [hashicorp/nomad#2026]
Nomad v0.5.0
rkt Version: 1.19.0

If a task's `trust_prefix` is not specified, the rkt driver adds `--insecure-options=all` to the `run` command.  This is very broad and disables many capabilities that probably aren't intended.  From `rkt run --help`:
```
      --insecure-options=none		comma-separated list of security features to disable. Allowed values: "none", "image", "tls", "ondisk", "http", "pubkey", "capabilities", "paths", "seccomp", "all-fetch", "all-run", "all"
```
The various options are documented [here](https://coreos.com/rkt/docs/latest/commands.html).
`seccomp` and `capabilities` should not be disabled when trust is not enabled.

**Comment by blalor** [comment]
Unless someone beats me to this, I think I can tackle the fix.  I'm considering exposing more of the rkt cli arguments to the driver, beyond the scope of this particular issue.  But for this issue, I'm thinking the following might work well:

```hcl
config {
    image = "example.com/base/go-server:3cfc9529-55"
    insecure_options = ["image","ondisk"]
}
```

The `insecure_options` would be orthogonal to any trust settings, which opens the door to allowing the operator to manage trust outside of Nomad, but would require the author of the jobspec to specify the appropriate options.

**Comment by dadgar** [comment]
@blalor Sounds good! Love to see some community love for the Rkt driver!

**Comment by blalor** [comment]
Somewhat orthogonal to this specific issue, are there guidelines on changing the schema of a driver's config?  Trying to think ahead about the consequences of adding or removing struct fields, or worst-case, what happens if a field changes type.

**Comment by dadgar** [comment]
Additions are fine to the schema. With renames a compatibility layer should be added for a release. Deletions are BC and need to be more carefully considered

**Comment by tgross** [comment]
Doing some issue cleanup... this issue was fixed by #2695. (And in any event rkt is deprecated at this point!)
